distributed/comm/ucx.py

```diff
@@ -175,7 +175,8 @@ def peer_address(self) -> str:
                 #  syncing the default stream will wait for other non-blocking CUDA streams.
                 # Note this is only sufficient if the memory being sent is not currently in use on
                 # non-blocking CUDA streams.
-                synchronize_stream(0)
+                if any([hasattr(f, "__cuda_array_interface__") for f in frames]):
```

Comment:
We collect this earlier on line 164 as well. Thoughts on just storing that to a variable so we can reuse it here?

Reply:
We could then do an analogous thing here as below.

```python
if is_cudas.any():
    ...
```